src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Web/Debugger/FlavorServerSettings.cs

```diff
@@ -0,0 +1,168 @@
+{
+    /// <summary>
+    /// Represents the settings stored in the FlavorProperties section of the project file
+    /// </summary>
+    internal class FlavorServerSettings
+    {
+        private static class PropertyNames
+        {
+            // project file tag names
+            public const string RootNode = "WebProjectProperties";
+            public const string UsingIIS = "UseIIS";
+            public const string ServerPort = "DevelopmentServerPort";
+            public const string ServerVPath = "DevelopmentServerVPath";
+            public const string IISUrl = "IISUrl";
+            public const string IISAppRootUrl = "IISAppRootUrl";
+            public const string OverrideIISAppRootUrl = "OverrideIISAppRootUrl";
+            public const string StartPageUrl = "StartPageUrl";
+            public const string DebugCurrentPage = "DebugCurrentPage";
+            public const string AutoAssignPort = "AutoAssignPort";
+            public const string EnableWcfTestClientForSVC = "EnableWcfTestClientForSVC";
+            public const string EnableWcfTestClientForSVCDefaultValue = "EnableWcfTestClientForSVCDefaultValue";
+            public const string SaveServerSettingsInUserFile = "SaveServerSettingsInUserFile";
+            public const string UseCustomServer = "UseCustomServer";
+            public const string CustomServerUrl = "CustomServerUrl";
+
+            // User File only settings which map to debug options
+            public const string AspNetDebugging = "AspNetDebugging";
+            public const string SilverlightDebugging = "SilverlightDebugging";
+            public const string NativeDebugging = "NativeDebugging";
+            public const string SQLDebugging = "SQLDebugging";
+            public const string NTLMAuth = "NTLMAuthentication";
+            public const string ServerDirectoryListing = "ServerDirectoryListing";
+            public const string UseClassLibDebugging = "UseClassLibraryDebugging";
+            public const string HideWebConfig = "HideWebConfig";
+            public const string AlwaysStartWebServerOnDebug = "AlwaysStartWebServerOnDebug";
+        };
+
+        public FlavorServerSettings() { }
+        public FlavorServerSettings(FlavorServerSettings other)
+        {
+            UseIIS = other.UseIIS;
+            SslPort = other.SslPort;
+            DevelopmentServerPort = other.DevelopmentServerPort;
+            DevelopmentServerVPath = other.DevelopmentServerVPath;
+            IISUrl = other.IISUrl;
+            IISAppRootOverrideUrl = other.IISAppRootOverrideUrl;
+            UseIISAppRootOverrideUrl = other.UseIISAppRootOverrideUrl;
+            CustomServerUrl = other.CustomServerUrl;
+            UseCustomServer = other.UseCustomServer;
+        }
+
+        public static FlavorServerSettings CreateFromXml(XElement flavorRoot)
+        {
+            var settings = new FlavorServerSettings();
+
+            var webPropertiesRoot = flavorRoot.Element(PropertyNames.RootNode);
```

Comment:
>flavorRoot [](start = 36, length = 10)

Would it make sense to return "settings" if flavorRoot is null as well?

Reply:
It does and I think that is correct since empty settings will be returned